dlp/api/Snippets.Tests/DeidentifyTableWithInfoTypesTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License
+
+using Google.Cloud.Dlp.V2;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace GoogleCloudSamples
+{
+    public class DeidentifyTableWithInfoTypesTests : IClassFixture<DlpTestFixture>
+    {
+        private DlpTestFixture _fixture;
+        public DeidentifyTableWithInfoTypesTests(DlpTestFixture fixture) => _fixture = fixture;
+
+        [Fact]
+        public void TestDeidentifyTableWithInfoTypes()
+        {
+            var row1 = new Value[] { new Value { StringValue = "101" }, new Value { StringValue = "Charles Dickens" }, new Value { StringValue = "95" }, new Value { StringValue = "Charles Dickens name was a curse invented by Shakespeare." } };
+            var row2 = new Value[] { new Value { StringValue = "22" }, new Value { StringValue = "Jane Austin" }, new Value { StringValue = "21" }, new Value { StringValue = "There are 14 kisses in Jane Austen's novels." } };
+            var row3 = new Value[] { new Value { StringValue = "55" }, new Value { StringValue = "Mark Twain" }, new Value { StringValue = "75" }, new Value { StringValue = "Mark Twain loved cats." } };
+
+            var table = new Table
+            {
+                Headers = { new FieldId { Name = "AGE" }, new FieldId { Name = "PATIENT" }, new FieldId { Name = "HAPPINESS SCORE" }, new FieldId { Name = "FACTOID" } },
+                Rows =
+                {
+                    new Table.Types.Row { Values = { row1 } },
+                    new Table.Types.Row { Values = { row2 } },
+                    new Table.Types.Row { Values = { row3 } }
+                }
+            };
+            var infoTypes = new InfoType[] { new InfoType { Name = "PERSON_NAME" } };
+            var result = DeidentifyTableWithInfoTypes.DeidentifyTable(_fixture.ProjectId, table, infoTypes);
+            Assert.NotEqual(JsonConvert.SerializeObject(table), JsonConvert.SerializeObject(result));
```

Comment:
![image](https://user-images.githubusercontent.com/77317274/232729667-975c6ab9-b314-4e8c-9539-d48a08b1091d.png)

like this or any other way? (will separate out the lines)

Reply:
This thread is similar to the one above, in that something specific about deidentification needs to be checked. The DLP folks may better advice with exactly what.

As for the how, yes, that's a way, and it's fine in this case, although I would prefer something more structured like:

```suggestion
            Assert.DoesNotContains(table, row => ....);
```
But yes, something like what you are doing is fine in this case, because it's harder to get to the values with these types.

Reply:
Its very hard to dive deeper in table object.

Reply:
Should I go with the snippet I have shared above? To check whether passed name is there or not

Reply:
Yep, I think that should be fine, it is similar to what you are doing in other tests.

Reply:
Okay Thanks. Let me change the other PRs also.